### ISSUE TITLE
Add back drag packing behavior from v1

### ISSUE THREAD

**Issue by desandro** [metafizzy/packery#353]
Packery v2 changed packing behavior. I'm interested to hear if anyone would like an option to use Packery v1 drag behavior in Packery v2.

+ **Packery v1** items will all be packed around dragged item. This is useful if you want the tightest arrangement of items.
+ **Packery v2** items maintain horizontal position, only shifting up or down to make space for dragged item. This is useful if you want less movement while dragging items. It's easier to keep track of how items will move.

![packery-shift1](https://cloud.githubusercontent.com/assets/85566/13862442/c15b907e-ec69-11e5-9e9f-da7e21fc1857.png)

**+1 or add a :+1: reaction** if you'd like to see Packery v1 drag packing behavior be added back to Packery v2.

**Comment by lgm42** [comment]
+1.
I need, behavior like flow layout panel in desktop development.

**Comment by Adryi13** [comment]
Can't we have both behaviors? If there is a way to toggle between them i would like to know. Thanks!

**Comment by cbtdev** [comment]
I'm pretty sure that is the intention. 

As desandro said:

> I'm interested to hear if anyone would like an ***option*** to use Packery v1 drag behavior in Packery v2.

**Comment by andrey-lizunov** [comment]
Hope that this will be implemented and not shelved.
In my case, `layout` method doesn't solve an issue(

**Comment by WilliamSWoodruff** [comment]
@desandro I'm interested in this please!

**Comment by WilliamSWoodruff** [comment]
What I'm really interested in is having v1-like predictable reordering with ability for variable-height elements, all same width. Is that possible?

**Comment by metaman** [comment]
:+1: Definitely would like to see this option back in please. Our items are all the same size so it makes sense for our users to be able to swap horizontally a <-> b and vertically a <-> f

**Comment by davidyarham** [comment]
I agree with @metaman. When dragging an item left to right then my brain would expect the right most to slide into its place, and also when dragging down then I would expect the underneath to slide into place.

Similar to a Slide Puzzle: http://s.hswstatic.com/gif/sliding-puzzles-1.jpg

**Comment by yairEO** [comment]
Is there an ETA for this? seems like a very important issue. I am very eager to see this done :)

**Comment by UranusCEO** [comment]
We really this function~ Anyone have a quick fix to it rather than waiting for the author?

**Comment by jeffaltman** [comment]
+1

**Comment by jeffaltman** [comment]
+1

**Comment by WeArePercipio** [comment]
+ 1

**Comment by Grey007** [comment]
What is the status of this feature?

**Comment by Eldair** [comment]
Any news on this?

**Comment by desandro** [comment]
This feature is still being considered. Adding another layout algorithm into Packery makes it more complex. So I'd like to be careful.

**Comment by andrey-lizunov** [comment]
I was just hoping that this feature will be implemented, otherwise I have to choose some other decision, because the current behaviour is just too weird in my case. I didn't expect this in the beginning, when choosing the library and thought that it would be the same as in Packery v1.
That's why I need some news about it. If you have some plans to implement it then I can wait, but if not, then I should find some other library for my project.

**Comment by jeffaltman** [comment]
+1

**Comment by gorkma** [comment]
+1

**Comment by kaisermann** [comment]
That would be really good. I use packery for arranging photo grids and the v2 behaviour is not intuitive (and produces """wrong""" results) for my current use case.

**Comment by jano42** [comment]
+1

**Comment by jano42** [comment]
It would be nice to hace the choice on the drag behavior !

**Comment by infabo** [comment]
+1

**Comment by infabo** [comment]
This is like pest and cholera. v1 has the good drag behaviour, but dragging events are buggy and broken. v2 has working drag events, but the drag behaviour is like "wtf".
No need to be polite, because this project looks dead to me and I don't expect any progress on this issue anyway.

Have a nice day!

**Comment by kaisermann** [comment]
@infabo Even if you consider the project "dead" there's no need to be aggressive. That only makes you another internet jerk. If you really think the project's dead you can fork it and bring it "back to life". 

Don't cry, commit.

**Comment by desandro** [comment]
@infabo Packery is not dead. I'm considering adding this feature back as its gained a lot of interest. 
Adding features makes projects more complex. So I'm always careful with what features get added. I appreciate your concern and energy with the project. I'm sorry if this issue has taken too long. Alas, I am but one person maintaining all of Metafizzy's projects.

**Comment by LEEVE** [comment]
With all due respect, the old behavior is illogical to me. 
Scenario 1: If you were to move A to column 3 I understand the logic behind that when C moves to the left one column it pushes the other ones out of the way and F ends up in column 3!??! In essence you moved both A and F from column 1 to 3 with one move. I would instead have B& C slide to the left, E pushes down to make room for C above it and F remains where it is the same with D&G (staying in their existing spots)
Scenario 2 is more predictable that it just pushes the existing cells down.
Scenario 3 would be the newer version that does what I outlined in my first paragraph.
So my vote would be Forget v1 & v2 and do v3 instead :)

**Comment by zenati** [comment]
Use https://github.com/haltu/muuri, simple to use and drag behaviour works like a charm.

**Comment by dodul** [comment]
It seems we can no longer place bigger containers under smaller ones. We can place smaller ones under the bigger ones or the same sized ones below one another. Please check the attached screenshot. I tried it from:  https://packery.metafizzy.co/draggable.html

![packery](https://user-images.githubusercontent.com/13984819/32008391-56a9111a-b969-11e7-8780-8375974beb3a.png)

It would be very useful if we can have the ability to place bigger containers under the small ones.

**Comment by geraldo** [comment]
Thanks for your efforts to keep maintained isotope and packery. I would like to know if you still consider implementing this feature, and if yes, WHEN? Just to know if I should switch to another library or not...

**Comment by LEEVE** [comment]
Thanks for the heads up. Feel free to switch I will switch along as well.

Sent from Mail for Windows 10

**Comment by pchekriy** [comment]
Hi, there's news?

**Comment by pchekriy** [comment]
@desandro hello, there's news about horizontal in v2?
